viewing_party/party.py

```diff
@@ -1,23 +1,141 @@
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_to_watch):
+    for i in range(len(user_data["watchlist"])):
+        if movie_to_watch in user_data["watchlist"][i]["title"]:
+            del user_data["watchlist"][i]
+            user_data["watched"].append(movie_to_watch)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
 
+    total_rating = 0
+    count = len(user_data["watched"])
+    if user_data["watched"]:
+        for i in range(count):
+            total_rating += user_data["watched"][i]["rating"]
+        return total_rating/count
+    else:
+        return 0
+
+def get_most_watched_genre(user_data):
+    rating = {}
+    if not user_data["watched"]:
+        return None
+    else:
+        for i in range(len(user_data["watched"])):
+            genre_name = user_data["watched"][i]["genre"]
+            if genre_name not in rating:
+                rating[genre_name] = 1
+            else: 
+                rating[genre_name] += 1
+            popular = [(value, key) for key, value in rating.items()]
+        return max(popular)[1]
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_friend_watched_movies(user_data):
+    friends_watched_movies = []
+    for friens_dict in user_data["friends"]: 
+        # print(f"{item=}") 
+        for friend_list in friens_dict["watched"]:
+            friends_watched_movies.append(friend_list)
+    return friends_watched_movies
+
+def get_unique_watched(user_data):
+    my_unique_movies = []
+    friends_watched_movies = get_friend_watched_movies(user_data)
+    # .append() seems more efficient
+    for my_movie in user_data["watched"]:
+        if my_movie not in friends_watched_movies:
+            my_unique_movies.append(my_movie)
+    return my_unique_movies
+    # .remove() method:
+    # for my_movie in user_data["watched"]:
+    #     my_unique_movies.append(my_movie)
+    # # print(f"{my_unique_movies=}")
+    # for item in friends_watched_movies:
+    #     if item in my_unique_movies:
+    #         my_unique_movies.remove(item)
+    # # print(f"{my_unique_movies=}")
+    # return my_unique_movies
+    
+def get_friends_unique_watched(user_data): 
+    friends_watched_movies = get_friend_watched_movies(user_data)
+    my_watched_movies = user_data["watched"]
+    friends_unique_movies = []
+
+    for friends_item in friends_watched_movies: 
+        # print(f"{item=}") 
+        if friends_item not in friends_unique_movies and friends_item not in my_watched_movies:
+            friends_unique_movies.append(friends_item)
+    return friends_unique_movies

```

Comment:
👍  Great use of a combined conditional expression for checking if the friends movie was in `friends_unique_movies` and `my_watched_movies`.